File: projecteuler4.cpp

#include <iostream>

using namespace std;
 
int res = 1;
 
int gcd(int a,int b) {
  
   if(b == 0)
      return a;
   else return gcd(b,a%b);    
}
 
void lcm(int arr[], int n) {
 
    if(n == 0) return;
    res = res*arr[n-1] / gcd(res, arr[n-1]);
    lcm(arr, n-1);
}
 
int main() {
     
    int arr[40];
    int n,t;
    cin>>t;
    for(int i = 0;i < t;i++)
    {
      res = 1;
      cin>>n;
      for(int j = 1;j<=n;j++)
      {
           arr[j-1] = j;
      }
      lcm(arr, n);
      cout << res << endl;
    }
  return 0;
}

File: projecteuler4.py
import sys

def gcd(a,b):
	if b==0:
		return a;
	else:
		return gcd(b,a%b)

def lcm(list,length):
    if(length == 0):
    	return 
	res = res*list[length-1] / gcd(res,list[length-1])
	lcm(list,length-1)

      

t = int(raw_input())

for i in range(1,t+1):
	res = 1
	l = []
	n = int(raw_input())
	for j in range(1,n+1):
		l.append(j)

	size = len(l)
	lcm(l,size)
	print res


    
    